tests/exception_test.py

```diff
@@ -0,0 +1,70 @@
+import sys
+from typing import TYPE_CHECKING
+
+from tornado.httpserver import HTTPRequest
+
+from grouper.plugin.base import BasePlugin
+from grouper.plugin.proxy import PluginProxy
+
+if TYPE_CHECKING:
+    from types import TracebackType
+    from typing import Optional, Type
+
+
+class RandomException(Exception):
+    pass
+
+
+class ExceptionLoggerTestPlugin(BasePlugin):
+    def __init__(self):
+        # type: () -> None
+        self.request = None  # type: Optional[HTTPRequest]
+        self.status = None  # type: Optional[int]
+        self.exc_type = None  # type: Optional[Type[BaseException]]
+        self.exc_value = None  # type: Optional[BaseException]
+        self.exc_tb = None  # type: Optional[TracebackType]
+
+    def log_exception(
+        self,
+        request,  # type: Optional[HTTPRequest]
+        status,  # type: Optional[int]
+        exc_type,  # type: Optional[Type[BaseException]]
+        exc_value,  # type: Optional[BaseException]
+        exc_tb,  # type: Optional[TracebackType]
+    ):
+        # type: (...) -> None
+        self.request = request
+        self.status = status
+        self.exc_type = exc_type
+        self.exc_value = exc_value
+        self.exc_tb = exc_tb
+
+
+def test_exception_plugin():
```

Comment:
Is anything specific to exception logging actually being tested here? It seems that there's only one piece of logic being run that isn't defined in this file, which is how `proxy.log_exception` does in fact call `plugin.log_exception` for the one passed in `plugin`.

Reply:
It isn't currently. :( I'd like it to really test exception handling, but haven't yet been able to figure out a good way to do so. The name of the test is therefore somewhat aspirational.